options.go

```diff
@@ -94,3 +94,27 @@ func (o *FetchOptions) Validate() error {
 
 	return nil
 }
+
+// PushOptions describe how a push should be performed.
+type PushOptions struct {
+	// RemoteName is the name of the remote to be pushed to.
+	RemoteName string
+	// RefSpecs specify what destination ref to update with what source
+	// object. A refspec with empty src can be used to delete a reference.
+	RefSpecs []config.RefSpec
+}
+
+// Validate validate the fields and set the default values
```

Comment:
I'm following the same convention that we are using for a lot of objects in go-git (check other Validate methods). We can discuss changing them as a separate issue, not related to this PR.